.github/workflows/package-windows.yml

```diff
@@ -0,0 +1,123 @@
+name: package-windows
+
+# This workflow will be triggered on:
+# - push on the main branch
+# - push on the releases branches
+# - tag push matching the version pattern
+on:
+  pull_request:
+    paths:
+      - .github/workflows/package-windows.yml
+  push:
+    branches:
+      - master
+      - "[0-9]+.[0-9]+"
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+
+env:
+  python-version: "3.9"
+
+jobs:
+  package-windows-build-wheel:
+    runs-on: windows-2022
+    env:
+      # Need feature from poetry 1.2.x that aren't present in 1.1.x
+      poetry-version: 1.2.0b3
+    outputs:
+      wheel-version: ${{ steps.wheel-version.outputs.version }}
+    steps:
+      - name: Install poetry
+        shell: bash
+        run: |
+          export POETRY_HOME=${APPDATA}/.poetry
+          curl -sSL https://install.python-poetry.org \
+            | python3 - --version=${{ env.poetry-version }}
+          echo "${APPDATA}\.poetry\bin" >> "$GITHUB_PATH"
+
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # pin v3.0.2
+
+      - uses: actions/setup-python@d09bd5e6005b175076f227b13d9730d56e9dcfcb  # pin v4.0.0
+        with:
+          python-version: ${{ env.python-version }}
+          cache: poetry
+
+      - name: Get wheel version
+        id: wheel-version
+        shell: bash
+        run: |
+          set -e
+          # Retrieve Parsec version and customize it if we are not building on a tag
+          # See https://www.python.org/dev/peps/pep-0440/#local-version-identifiers
+          if ([ "${{ github.ref_type }}" == "tag" ])
+          then
+            # Use tag name as version
+            VERSION=`git describe --tag`
+          else
+            # Use `<tag_name>+dev<commit_hash>` as version
+            VERSION=`sed -n 's/^__version__ = "\\(.*\\)"$/\\1/p' parsec/_version.py`
+            COMMIT=`git rev-parse --verify --short HEAD`
+            VERSION="$VERSION.$COMMIT"
+            # Customize version with tag
+            sed -i "s#^\\(__version__ = \\).*\$#\\1'$VERSION'#" parsec/_version.py
+            test $(grep -E 'version = ' -o pyproject.toml | wc -l) -eq 1  # Sanity check for the next sed
+            sed -i "s#^\\(version = \\).*\$#\\1'$VERSION'#" pyproject.toml
+          fi
+
+          echo "::warning title=Wheel version::$VERSION"
+          echo "::set-output name=version::$VERSION"
+
+      - name: Create output dir
```

Comment:
With `set -eux` there is no need for such fine separation given the executed line is printed

So we can merge `Get wheel version` -> `Debug dist dir` all together into a single `Generate Wheel` (and we would avoid the `::set-output` and have a good old env variable instead ^^)

Reply:
Yup, I agree that having `Get wheel version` in it own step bring redability

however `Create output dir` + `Set file for wheel version` + `Build wheel` + `Debug dist dir` are just 4 one-liners that should be glued together (with a couple of comments if necessary)